src/windows/gfx/d3d11-ctx.c

```diff
@@ -29,11 +29,24 @@ struct d3d11_ctx {
 	uint32_t width;
 	uint32_t height;
 	MTY_Renderer *renderer;
+	DXGI_FORMAT format;
+	DXGI_FORMAT format_new;
+	DXGI_COLOR_SPACE_TYPE colorspace;
+	DXGI_COLOR_SPACE_TYPE colorspace_new;
 	ID3D11Device *device;
 	ID3D11DeviceContext *context;
 	ID3D11Texture2D *back_buffer;
 	IDXGISwapChain2 *swap_chain2;
+	IDXGISwapChain3 *swap_chain3;
+	IDXGISwapChain4 *swap_chain4;
+	IDXGIFactory1 *factory1;
 	HANDLE waitable;
+	bool hdr_init;
+	bool hdr_supported;
+	bool hdr;
+	bool composite_ui;
```

Comment:
Can't the usage not be replaced with just ->hdr?

Reply:
Reviewed my code and now I remember why I added that bool. In short, you cannot remove it and just use ->hdr. Here's why:

There are three expected scenarios with drawing the app:
1) Draw HDR quad. Draw SDR UI on top of HDR Quad (special composite step). Present.
2) Draw SDR quad. Draw SDR UI on top of SDR Quad. Present.
3) Only draw SDR UI. Present.

There's no clean external way to know when the scenario has changed from 1 (or 2) to 3. The composite_ui flag lets me efficiently know whether I'm in scenario 1/2 or 3. This information is not possessed by the ctx->hdr flag, which only can tell you whether you're in scenario 1 or 2.